### PR TITLE
Initial deployment to App Engine

### PR DESCRIPTION
**Pull request by mmunozvaltierra**
To correctly deploy to app engine I had to reorganize the files. In the root directory we can now call "npm run dev" to begin both servers at once when developing. /build files can be ignored as they were generated by react

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DB_HOST=34.122.81.101
+DB_DATABASE=shoplocal
+DB_USER=demo
+DB_PASS=password
```
Comment (nitinms-sps-program) [inline .env]: Please do not add credentials into database.

File: app.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+runtime: nodejs
+
+handlers:
+  - url: /
+    static_files: shop-local/build/index.html
+    upload: shop-local/build/index.html
+  - url: /
+    static_dir: shop-local/build
```
Comment (nitinms-sps-program) [inline app.yaml]: Can we fix the entries in app.yaml?

File: index.js [hunk 3]
```diff
@@ -93,6 +96,11 @@ app.get("/get-store-id", (req, res) => {
 
 });
 
+// All other GET requests not handled before will return index.html
+app.get('*', (req, res) => {
```
Comment (nitinms-sps-program) [inline index.js]: This should be /*

File: index.js [hunk 4]
```diff
@@ -93,6 +96,11 @@ app.get("/get-store-id", (req, res) => {
...
+    res.sendFile(path.resolve(__dirname, '/shop-local/build', 'index.html'));
```
Comment (nitinms-sps-program) [inline index.js]: '/shop-local/build/index.html'

File: package.json [hunk 5]
```diff
@@ -4,13 +4,16 @@
   "description": "",
   "main": "db-connect.js",
   "scripts": {
-    "start": "node database/index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "node index.js",
```
Comment (nitinms-sps-program) [inline package.json]: Also

"server": "node index.js",

File: shop-local/build/asset-manifest.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (nitinms-sps-program) [inline shop-local/build/asset-manifest.json]: Files from /shop-local/build
should not be added to GitHub as they are built using npm and file names will change everything the build is done.

File: app.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+runtime: nodejs14
+
+handlers:
```
Comment (nitinms-sps-program) [inline app.yaml:3]: handlers:
  - url: /get-store
     script: auto
  - url: /list-stores
     script: auto
  - url: /get-store-id
     script: auto

File: app.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+  - url: /
+    static_files: shop-local/build/index.html
+    upload: shop-local/build/index.html
+  - url: /
+    static_dir: shop-local/build
+
+env_variables:
+  PORT: 8080
```
Comment (nitinms-sps-program) [inline app.yaml]: DB_HOST: <IP>
  DB_DATABASE: shoplocal
  DB_USER: demo
  DB_PASS: <PASSWORD>
  DB_INSTANCE_NAME: /cloudsql/mmunozvaltierra-sps-spring21:us-central1:shop-local

File: index.js [hunk 9]
```diff
@@ -2,12 +2,15 @@ require("dotenv").config();
...
 
 const PORT = process.env.PORT || 3001;
 
 const app = express();
 let jsonResponse = null;
 
+app.use(express.static(path.resolve(__dirname, '/shop-local/build')));
+
 //Configure connection
 let connection = mysql.createConnection({
```
Comment (nitinms-sps-program) [inline index.js:15]: database: process.env.DB_DATABASE,
socketPath: process.env.DB_INSTANCE_NAME

**Review by nitinms-sps-program — COMMENTED** [review]
There were a few fixes outside the PR:

1) Needed to start the Cloud SQL Server.

2) Needed to add spring21-sps-44@... account into IAM settings for Mauricio's project to be able to access Cloud SQL.

**Review by Caortega4 — APPROVED** [review]

**Pull request merged**